python/graphstorm/model/edge_gnn.py

```diff
@@ -180,12 +180,33 @@ def edge_mini_batch_gnn_predict(model, loader, return_proba=True, return_label=F
     labels = {}
     model.eval()
 
+    len_loader = max_num_batch = len(list(loader))
+    tensor = th.tensor([len_loader], device=device)
+    if is_distributed():
+        th.distributed.all_reduce(tensor, op=th.distributed.ReduceOp.MAX)
+        max_num_batch = tensor[0]
+
+    dataloader_iter = iter(loader)
+
     with th.no_grad():
-        for input_nodes, target_edge_graph, blocks in loader:
+        # To use WholeGraph for feature featching, dataloaders from different
+        # trainers must iterate through the same number of iterations as WholeGraph
+        # does not support imbalanced batch numbers across processes/trainers
+        # TODO (IN): Fix dataloader to have the same number of minibatches
+        for iter_l in range(max_num_batch):
+            if iter_l < len_loader:
+                input_nodes, target_edge_graph, blocks = next(dataloader_iter)
+            else:
+                for ntype in g.ntypes:
+                    input_nodes[ntype] = target_edge_graph[ntype] = th.empty((0,), dtype=g.idtype)
```

Comment:
```suggestion
                    input_nodes[ntype] = th.empty((0,), dtype=g.idtype)
```
We do not need to init target_edge_graph as if blocks is None, we will skip the steps after L209.

Reply:
Fixed.